modules/src/ics03_connection/handler/conn_open_init.rs

```diff
@@ -17,9 +17,7 @@ pub(crate) fn process(
     let mut output = HandlerOutput::builder();
 
     // An IBC client running on the local (host) chain should exist.
-    if ctx.client_state(msg.client_id()).is_none() {
-        return Err(Error::missing_client(msg.client_id().clone()));
-    }
+    ctx.client_state(msg.client_id())?;
```

Comment:
We could delete `Ics03Error::MissingClient` and reuse `Ics03Error::Ics02Client` variant to propagate client errors.

Reply:
This would also necessitate changes like the following ->
context.rs:701
```rust
ClientReader::client_state(self, client_id).map_err(Ics03Error::ics02_client)
```